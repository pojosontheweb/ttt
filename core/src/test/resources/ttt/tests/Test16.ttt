<%@ page import="java.util.List" %>
<%@ page contentType="text/html" %>
<%--
    JSP comment can span
    appear anywhere
    and are ignored when template gets compiled
--%>
<%@ page extends="java.io.Serializable" %>
<%!
    List<? extends Number> foo;
    int bar;
%>
<div class="foo">
    <ul>
    <% for(Number n : foo) { %>
        <li>i equals <%= n %> </li>
    <% } %>
    </ul>
</div>