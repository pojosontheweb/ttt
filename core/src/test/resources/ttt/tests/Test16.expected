package ttt.tests;

import java.util.List;

public class Test16 extends com.pojosontheweb.ttt.Template implements java.io.Serializable {

	private final List<? extends Number> foo;
	private final int bar;

	/**
	 * Creates an instance of this template.
	 *
	 */
	public Test16(List<? extends Number> foo, int bar) {
		this.foo = foo;
		this.bar = bar;
	}

	@Override
	public String getContentType() {
		return "text/html";
	}

	@Override
	public void doRender(com.pojosontheweb.ttt.TttWriter out) throws Exception {
		write(out, "<div class=\"foo\">\n    <ul>\n    " );
		for(Number n : foo) { 
		write(out, "\n        <li>i equals " );
		write(out,  n  );
		write(out, " </li>\n    " );
		} 
		write(out, "\n    </ul>\n</div>" );
	}
}
